src/commands/synthetics/__tests__/reporters/junit.test.ts

```diff
@@ -253,14 +322,202 @@ describe('Junit reporter', () => {
+      getTestCase(browserTest, '10', {location: locations[0], timedOut: true}),
+      // Browser test, location 2, (timed out)
+      getTestCase(browserTest, '11', {location: locations[1], timedOut: true}),
+      // Browser test, location 2, (timed out)
+      getTestCase(browserTest, '12', {location: locations[1], timedOut: true}),
+    ]
+
+    const caseNames = testCases.map((testCase) => testCase.$.name)
+
+    const uniqueCaseNames = [...new Set(caseNames)]
+    expect(uniqueCaseNames.length).toEqual(caseNames.length)
+
+    expect(caseNames).toStrictEqual([
+      // API tests.
+      'Test name - id: aaa-aaa-aaa - location: aws:eu-central-1',
+      'Test name - id: aaa-aaa-aaa - location: aws:eu-central-2',
+      'Test name - id: aaa-aaa-aaa - location: aws:eu-central-1 - result id: 3 (not yet received)',
+      'Test name - id: aaa-aaa-aaa - location: aws:eu-central-2 - result id: 4 (not yet received)',
+
+      // Browser tests.
+      'Test name - id: bbb-bbb-bbb - location: aws:eu-central-1 - device: chrome.laptop_large',
+      'Test name - id: bbb-bbb-bbb - location: aws:eu-central-1 - device: firefox.laptop_large',
+      'Test name - id: bbb-bbb-bbb - location: aws:eu-central-2 - device: chrome.laptop_large',
+      'Test name - id: bbb-bbb-bbb - location: aws:eu-central-2 - device: firefox.laptop_large',
+      'Test name - id: bbb-bbb-bbb - location: aws:eu-central-1 - result id: 9 (not yet received)',
+      'Test name - id: bbb-bbb-bbb - location: aws:eu-central-1 - result id: 10 (not yet received)',
+      'Test name - id: bbb-bbb-bbb - location: aws:eu-central-2 - result id: 11 (not yet received)',
+      'Test name - id: bbb-bbb-bbb - location: aws:eu-central-2 - result id: 12 (not yet received)',
+    ])
+  })
+
+  it('all columns are filled in the test report table', () => {
+    const baseResult = getFailedBrowserResult()
+    const result: Result = {
+      ...baseResult,
+      test: {...baseResult.test, suite: 'tests.json'},
+    }
+
+    reporter['resultEnd'](result, '')
+
+    const testCase = reporter['json'].testsuites.testsuite[0].testcase[0]
+
+    const name = 'Test name - id: abc-def-ghi - location: Location name - device: chrome.laptop_large'
+    const failure = {
+      $: {
+        allowFailure: 'false',
+        step: 'Assert',
+        type: 'assertion',
+      },
+      _: 'Step failure',
+    }
+
+    expect(testCase.$).toHaveProperty('classname', 'tests.json') // Suite
+    expect(testCase.$).toHaveProperty('name', name) // Name
+    expect(testCase.$).toHaveProperty('file', 'tests.json') // Filename
+    expect(testCase.failure).toStrictEqual([failure]) // Status
+    expect(testCase.$).toHaveProperty('time', 20) // Duration
+  })
+
+  it('the icon in the Status column is correct (blocking vs. non-blocking)', () => {
+    // Mimics how GitLab chooses the Status icon.
+    const getStatusIcon = (testCase: XMLTestCase) => {
+      if (testCase.failure.length > 0) {
+        return '❌'
```

Comment:
We could use plain text for these ones (e.g. `'error'` and `'skipped'`) but emojis add some colors to these tests 🌈

Reply:
Yes, I like it! It's more visual than "error"/"failure" and it really shows what GitLab would display in the UI.